tests/MongoDB.Driver.Tests/Specifications/server-selection/ServerSelectionTests.cs

```diff
@@ -0,0 +1,93 @@
+/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.server_selection
+{
+    public class ServerSelectionTest
+    {
+        [Theory]
+        [ParameterAttributeData]
+        public void ReadPreference_should_not_be_sent_to_standalone_server(
+            [Values(false, true)] bool async)
+        {
+            var eventCapturer = new EventCapturer().Capture<CommandStartedEvent>(e => e.CommandName.Equals("find") || e.CommandName.Equals("$query"));
```

Comment:
How can command name ever equal `$query`?

If it does there is probably a bug somewhere.

Reply:
There may be a bug. In the evergreen test for 3.2 sharded cluster no-auth test, I was using exceptions to debug and this was the command that came up: `{ "$query" : { "find" : "testcollection", "filter" : { "x" : 2 } }, "$readPreference" : { "mode" : "primaryPreferred" } }`. The patch is [here](https://evergreen.mongodb.com/task/dot_net_driver_unsecure_tests__version~3.2_os~windows_64_topology~sharded_cluster_auth~noauth_ssl~nossl_test_patch_16ba84a88792c695d3135a3ec7a687e0a0716b5a_5d42f783d1fe07102ee6a681_19_08_01_14_30_28), if you'd like to see.

Reply:
What's also confusing is that earlier versions (2.6 and 3.0 on evergreen) have the mock command message, while 3.2 and 3.4 seem to have the `$query` command.

Reply:
Hmm... It could be worth setting a breakpoint [here](https://github.com/mongodb/mongo-csharp-driver/blob/1c5493fa4c9593abaf15a2d8464a762baa35045a/src/MongoDB.Driver.Core/Core/Connections/CommandEventHelper.cs#L592) and seeing how we're getting a `$query` as the command (or if we're even getting into that function at all)

Reply:
This comment is still pending.

How can command name ever equal `$query`?

Reply:
This will be resolved in another ticket, not in this PR (see discussion below).